client/src/featureform/client.py

```diff
@@ -188,6 +192,135 @@ def close(self):
         """
         self.impl.close()
 
+    def add_trigger(self, trigger, resource):
+        """
+        Add a trigger to a resource
+
+        **Example:**
+        ```py title="definitions.py"
+        client.add_trigger("trigger_name", ("resource_name", "resource_variant, resource_type"))
+        ```
+
+        Args:
+            trigger(Union[str, TriggerResource]): The name of the trigger
+            resource(Union[tuple, FeatureColumnResource, TrainingSetVariant]): The name, variant and type of the resource
+        """
+        req = metadata_pb2.TriggerRequest()
+
+        trigger_req = metadata_pb2.Trigger()
+        if isinstance(trigger, str):
+            trigger_req.name = trigger
+        elif isinstance(trigger, TriggerResource):
+            trigger_req.name = trigger.name
+        else:
+            raise ValueError("Invalid trigger type")
+        req.trigger.CopyFrom(trigger_req)
+
+        resource_req = metadata_pb2.ResourceID()
+        if isinstance(resource, tuple):
```

Comment:
You probably want to add a check for the tuple length so that you don't run into an `IndexError` exception when indexing to get the name, variant and type. This will also help provide the use with actionable information to correct the mistake.